hack/shared-cluster.sh

```diff
@@ -1,63 +1,86 @@
 #!/bin/bash
-
+set -e
 # shared-cluster.sh is used to provide a repeatable production cluster create script
 
-LOCATION="westcentralus"
-CLUSTER="shared-cluster"
-RESOURCEGROUP="shared-cluster"
-CLUSTERRESOURCEGROUP="aro-shared-cluster"
 
 usage() {
 	echo -e "usage: ${0} <create|delete>"
 	exit 1
 }
 
+check_env_set(){
+  if [[ -z "${!1}" ]]; then
+    echo "$1 is a required ENV but is unset."
+    exit 1
+  fi
+}
+
+check_env_set SHARED_CLUSTER_SHARED_CLUSTER_LOCATION
+check_env_set SHARED_CLUSTER_CLUSTER_NAME
+check_env_set SHARED_CLUSTER_RESOURCE_GROUP_NAME
+check_env_set SHARED_CLUSTER_CLUSTER_RESOURCE_GROUP_NAME
+
+
 if [ "$#" -ne 1 ]; then
 	usage
 
 elif [[ $1 == create ]]; then
     echo "creating resource group and network"
-    az group create --name $RESOURCEGROUP --location $LOCATION
+    az group create \
+    --name $SHARED_CLUSTER_RESOURCE_GROUP_NAME \
+    --location $SHARED_CLUSTER_LOCATION \
+    --tags persist:true  # This tag stops the RG being cleaned up
```

Comment:
Nit - we don't need this tag in the shared cluster script since the purge pipeline (clean.go) will only clean up resource groups with `purge:true`, which is automatically applied to any cluster created by a dev RP, but not applied to any cluster created by a prod RP. Since the shared cluster is a prod cluster without `purge`, it doesn't need to be tagged with `persist`, the clean.go script will ignore it either way.

If the intent is to change that behavior and clean up both prod and dev clusters, maybe we can handle that + the shared cluster tagging in a separate PR?

Overall, I think that the `purge` vs `persist` tagging conventions we use are confusing and we should probably just use one or the other. This could be a future PR / enhancement / doc.

Reply:
> Overall, I think that the purge vs persist tagging conventions we use are confusing and we should probably just use one or the other. This could be a future PR / enhancement / doc.

Couldn't agree more. I added it here for what seemed like consistency to me i.e. by supporting both and not choosing either in this PR we can decide to drop to just one in the future "easier". Happy to drop the tag here if you feel strongly but IMO (fresh and mostly context free as it is ;) ) we should look at all this together and decide how much we actually care. 

I actually wrote up some notes on this whole topic [here](https://docs.google.com/document/d/1UKHAoSRt2qLU53at9kdYg_3B3BDQW39FCoqYbZ3yJ0o/edit) that you might find interesting.